libsnark/zk_proof_systems/plonk/tests/example.tcc

```diff
@@ -0,0 +1,922 @@
+/** @file
+ *****************************************************************************
+Instantiation of the test vector values from the Python implementation
+of the Plonk protocol.
+
+See example.hpp .
+
+ *****************************************************************************
+ * @author     This file is part of libff, developed by Clearmatics Ltd
+ *             (originally developed by SCIPR Lab) and contributors
+ *             (see AUTHORS).
+ * @copyright  MIT license (see LICENSE file)
+ *****************************************************************************/
+#ifndef __PLONK_EXAMPLE_CPP__
+#define __PLONK_EXAMPLE_CPP__
+
+namespace libsnark
+{
+
+template<typename ppT> plonk_example<ppT>::plonk_example()
```

Comment:
Is this function generic over the curve?  There seem to be a lot of hard-coded values which are unlikely to work in the context of other curves.

Reply:
Definitely not generic. The `plonk_example` class contains internal values from the execution of the Plonk protocol for a specific test circuit (described in [VB19] https://vitalik.ca/general/2019/09/22/plonk.html) for one specific curve (`BLS12-381`). These values are used as test vectors for debugging our Plonk implementation.

Therefore the `plonk_example()` function is definitely not generic over the curves and is specific to the `BLS12-381` curve. Would you suggest to remove its template-ization and add a check that we are working over the `BLS12-381` curve before calling `plonk_example`?

Reply:
Yes.  If it is specific to a single curve, let's remove the template param.  Where we have implementations for some but not all curves, we can probably declare the generic version and provide specialized implementations only for specific `pp`s.

Reply:
Ok, removed the template parameter `ppT` from the `plonk_example` class. Also added an exception check for `ppT == bls12_381_pp` before each creation of a new plonk_example object. Once the implementation is tested and is stable, both the example class and the exception checks will be removed.

Reply:
Do you mean that the code will not even compile for curves other than bls12-381? If so, then indeed the check is not necessary. I added it in order to avoid errors during runtime.